R/class-Link.R

```diff
@@ -83,10 +119,80 @@ check_link_list <- function(object) {
 #' @rdname LinkList-class
 #' @importFrom S4Vectors SimpleList
 #' @export
-LinkList <- setClass("LinkList", 
+setClass("LinkList", 
   contains = "SimpleList",
   prototype = prototype(
     elementType = "Link"
   ),
   validity = check_link_list
-)
\ No newline at end of file
+)
+
+#' Generate a LinkList
+#' 
+#' Generate a LinkList from an edgeList
+#' @param object Object containing data to be converted to a LinkList
+#' @param linkColorScheme Either 'none' or 'posneg'. If 'posneg', the link color will be based on the sign of the weight.
+#' @return LinkList
+#' @export
+#' @examples
+#' LinkList(data.frame(source='a',target='b'))
+setGeneric("LinkList", function(object, linkColorScheme = c('none', 'posneg')) standardGeneric("LinkList"), signature = c("object"))
+
+#' @export
+setMethod("LinkList", "data.frame", function(object = data.frame(source=character(),target=character()), linkColorScheme = c('none', 'posneg')) {
+  if (!isValidEdgeList(object)) {
+    stop(paste(errors, collapse = '\n'))
+  }
+
```

Comment:
i think i need more info here. what is it youre trying to prevent? when would we need to coerce data to fit a color scheme? 

if youd like i can put out a warning that the color scheme arg will be ignored if color is already set, and not set the color scheme slot. that seems reasonable enough to start.

Reply:
im starting to wonder if an edgeList should be allowed to have a color column. its completely ambiguous what the expected behavior should be in this case and i dont have a real use case for it i think. do you?

Reply:
I'm trying to prevent a Network from being created that has values in the link color slots that don't match the `linkColorScheme` slot. I like the warning idea, also we could just assign the `linkColorScheme` to be "custom" in the case where we're handed color when making a Network and not adding color via one of our setLinkColor (or whatever we called it) functions. 
No i don't have any near-term use cases. Eventually we might want a network with edges colored by the type of evidence used to assign that edge (ex. correlation, two-hybrid screen, etc), which would end up with us likely getting a list of link colors from somewhere else. But that's far away.

Reply:
ive just prevented custom color schemes for now, and figure we can revisit when we meet a use case